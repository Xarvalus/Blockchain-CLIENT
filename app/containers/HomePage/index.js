import * as React from 'react'
import { FormattedMessage } from 'react-intl'

import { Container } from 'reactstrap'
import { H1 } from 'components/Heading'

import messages from './messages'

const HomePage = (): React.Node => (
  <Container>
    <H1><FormattedMessage {...messages.header} /></H1>
  </Container>
)

export default HomePage
