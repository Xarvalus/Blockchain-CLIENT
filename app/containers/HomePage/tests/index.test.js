import React from 'react'
import { shallow } from 'enzyme'
import ShallowRenderer from 'react-test-renderer/shallow'

import { FormattedMessage } from 'react-intl'

import HomePage from '../index'
import messages from '../messages'

describe('<HomePage />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<HomePage />)
  })

  it('renders correctly', () => {
    const renderer = new ShallowRenderer()
    const snapshot = renderer
      .render(<HomePage />)

    expect(snapshot).toMatchSnapshot()
  })

  it('text matches localization', () => {
    expect(wrapper.find(FormattedMessage).props().defaultMessage).toBe(messages.header.defaultMessage)
  })
})
