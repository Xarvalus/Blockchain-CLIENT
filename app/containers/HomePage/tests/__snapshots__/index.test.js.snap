// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders correctly 1`] = `
<Container
  tag="div"
>
  <H1__H1>
    <FormattedMessage
      defaultMessage="Blockchain Client"
      id="app.containers.HomePage.header"
      values={Object {}}
    />
  </H1__H1>
</Container>
`;
